### PR TITLE
Adding Dockerfile

### PR DESCRIPTION
**Pull request by snehesht**
This PR adds instructions and necessary files required to build docker image.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+FROM ubuntu:16.04
+
+# Set the locale
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
```
Comment (Nikkau) [inline docker/Dockerfile]: You should squash the 3 `ENV` instructions into one to reduce layers number.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+RUN apt-get update && \
+    apt-get install -y wget && \
+    wget https://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb && \
+    dpkg -i erlang-solutions_1.0_all.deb && \
+    apt-get update && \
+    apt-get -y install esl-erlang && \
+    apt-get -y install elixir && \
+    apt-get -y install git
```
Comment (Nikkau) [inline docker/Dockerfile]: `ADD`can download things, no need to add a dependency for that.

You should squash all dependencies in one `apt-get install`.

You should clear cache after to reduce image size.

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+RUN git clone https://github.com/sergiotapia/magnetissimo /magnetissimo
```
Comment (Nikkau) [inline docker/Dockerfile]: You should use `COPY` to import local version to let people be able to test the current branch.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    mix ecto.migrate --force
+
+RUN wget https://deb.nodesource.com/setup_6.x && \
+    bash setup_6.x && \
+    apt-get update && \
+    apt-get install -y nodejs && \
+    apt-get install -y build-essential
+
+RUN cd /magnetissimo && \
+    npm install
```
Comment (Nikkau) [inline docker/Dockerfile]: All JS things should be removed.

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+EXPOSE 4000
+
+CMD cd /magnetissimo && mix phoenix.server
```
Comment (Nikkau) [inline docker/Dockerfile]: You should use `WORKDIR`if you need to change current directory.

File: docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    wget https://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb && \
+    dpkg -i erlang-solutions_1.0_all.deb && \
+    apt-get update && \
+    apt-get -y install esl-erlang && \
+    apt-get -y install elixir && \
+    apt-get -y install git
+
+RUN git clone https://github.com/sergiotapia/magnetissimo /magnetissimo
+
+ADD dev.exs /magnetissimo/config/dev.exs
```
Comment (Nikkau) [inline docker/Dockerfile]: You should use `COPY` instead.

File: docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+RUN cd /magnetissimo && \
+    mix local.hex --force && \
+    mix local.rebar --force && \
+    mix deps.get --force && \
+    mix ecto.create --force && \
+    mix ecto.migrate --force
```
Comment (Nikkau) [inline docker/Dockerfile]: Migrations are a run concern concern, not a build one, the building of the image shouldn't need a database running and you have no guarantee you will have the same one when you will launch the container.

**Comment by sergiotapia** [comment]
I'm not familiar with Docker at all, but this PR seems to be a bit more succinct and cleaner.

https://github.com/sergiotapia/magnetissimo/pull/41

Which PR is better @snehesht, this one or #41

**Comment by snehesht** [comment]
@sergiotapia PR #41 seems cleaner, you can close this one and merge PR #35

**Comment by sergiotapia** [comment]
@snehesht thanks for the heads up. Can you comment on what needs to be changed on the thread for #41? Thanks so much for your help

File: docker/Dockerfile (hunk 7 above)
Comment (snehesht) [inline docker/Dockerfile]: @Nikkau I'm not familiar with Elixer tools, please feel free to make necessary changes.

**Pull request closed without merging**